octobot_trading/personal_data/orders/groups/one_cancels_the_other_order_group.py

```diff
@@ -0,0 +1,57 @@
+#  Drakkar-Software OctoBot-Trading
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import octobot_trading.personal_data.orders.order_group as order_group
+import octobot_trading.errors as errors
+
+
+class OneCancelsTheOtherOrderGroup(order_group.OrderGroup):
+    """
+    OneCancelsTheOtherOrderGroup is linking orders together in the way that if any order of the group is filled
+    order cancelled, all the orders are cancelled.
+    Mostly used to pair stop and limit orders together in a stop-loss / take-profit setup
+    """
+
+    async def on_fill(self, filled_order, ignored_orders=None):
+        """
+        Called when an order referencing this group is filled
+        This is called right before updating portfolio for this filled order and the
+        order fill publication
+        :param filled_order: the filled order
+        :param ignored_orders: orders that should be ignored
+        """
+        if not self.enabled:
+            return
+        for order in self.get_group_open_orders():
+            if order is not filled_order:
+                await order.trader.cancel_order(order, ignored_order=filled_order)
+
+    async def on_cancel(self, cancelled_order, ignored_orders=None):
+        """
+        Called when an order referencing this group is cancelled
+        This is called before updating portfolio for this cancelled order and the
+        order cancel publication
+        :param cancelled_order: the cancelled order
+        :param ignored_orders: orders that should be ignored
+        """
+        if not self.enabled:
+            return
+        if ignored_orders and len(ignored_orders) > 1:
+            raise errors.OrderGroupTriggerArgumentError(f"ignored_orders supports at most 1 argument "
```

Comment:
Why?

Reply:
Because other ignored_orders (after 1st) will be ignored, I think we should not let it run.